### PR TITLE
Improvements to credential type documentation and validation bug fix.

### PR DESCRIPTION
**Pull request by mjspi**
### Change Description
Clarified the `title` and `description` fields in the credential type align to the ${cardTitle} and ${cardSubtitle} fields in the card_design_template.  Previously, the documentation incorrectly stated the `metadata.name` and `metadata.description` fields aligned to those card values.

Corrected the validation rules and test cases to align to the above also.

### Testing Results
All tests within the test suite passed.  Focus on the credential type tests also passed.
--- PASS: TestAccCredentialType_MetaData (0.64s)
=== CONT  TestAccCredentialTypeDataSource_NotFound
--- PASS: TestAccCredentialTypeDataSource_NotFound (0.73s)
=== CONT  TestAccCredentialTypeDataSource_ByIDFull
--- PASS: TestAccCredentialTypesDataSource_NotFound (6.80s)

#### Testing Shell Command

```shell
TF_LOG=WARN TF_LOG_PATH=`pwd`/output.log TF_ACC=1 go test -v -timeout 240s github.com/pingidentity/terraform-provider-pingone/internal/service/credentials
```

#### Testing Results

```shell
% TF_LOG=DEBUG TF_LOG_PATH=`pwd`/output.log TF_ACC=1 go test -v -timeout 240s github.com/pingidentity/terraform-provider-pingone/internal/service/credentials
=== RUN   TestAccCredentialIssuanceRuleDataSource_ByIDFull
=== PAUSE TestAccCredentialIssuanceRuleDataSource_ByIDFull
=== RUN   TestAccCredentialIssuanceRuleDataSource_NotFound
=== PAUSE TestAccCredentialIssuanceRuleDataSource_NotFound
=== RUN   TestAccCredentialIssuanceRuleDataSource_InvalidConfig
=== PAUSE TestAccCredentialIssuanceRuleDataSource_InvalidConfig
=== RUN   TestAccCredentialIssuerProfileDataSource_ByEnvironmentIDFull
=== PAUSE TestAccCredentialIssuerProfileDataSource_ByEnvironmentIDFull
=== RUN   TestAccCredentialIssuerProfileDataSource_NotFound
=== PAUSE TestAccCredentialIssuerProfileDataSource_NotFound
=== RUN   TestAccCredentialTypeDataSource_ByIDFull
=== PAUSE TestAccCredentialTypeDataSource_ByIDFull
=== RUN   TestAccCredentialTypeDataSource_NotFound
=== PAUSE TestAccCredentialTypeDataSource_NotFound
=== RUN   TestAccCredentialTypeDataSource_InvalidConfig
=== PAUSE TestAccCredentialTypeDataSource_InvalidConfig
=== RUN   TestAccCredentialTypesDataSource_NoFilter
=== PAUSE TestAccCredentialTypesDataSource_NoFilter
=== RUN   TestAccCredentialTypesDataSource_NotFound
=== PAUSE TestAccCredentialTypesDataSource_NotFound
=== RUN   TestAccDigitalWalletApplicationDataSource_ByIDFull
=== PAUSE TestAccDigitalWalletApplicationDataSource_ByIDFull
=== RUN   TestAccDigitalWalletApplicationDataSource_ByApplicationIDFull
=== PAUSE TestAccDigitalWalletApplicationDataSource_ByApplicationIDFull
=== RUN   TestAccDigitalWalletApplicationDataSource_ByNameFull
=== PAUSE TestAccDigitalWalletApplicationDataSource_ByNameFull
=== RUN   TestAccDigitalWalletApplicationDataSource_NotFound
=== PAUSE TestAccDigitalWalletApplicationDataSource_NotFound
=== RUN   TestAccDigitalWalletApplicationsDataSource_NoFilter
=== PAUSE TestAccDigitalWalletApplicationsDataSource_NoFilter
=== RUN   TestAccDigitalWalletApplicationsDataSource_NotFound
=== PAUSE TestAccDigitalWalletApplicationsDataSource_NotFound
=== RUN   TestAccCredentialIssuanceRule_Full
=== PAUSE TestAccCredentialIssuanceRule_Full
=== RUN   TestAccCredentialIssuanceRule_InvalidConfigs
=== PAUSE TestAccCredentialIssuanceRule_InvalidConfigs
=== RUN   TestAccCredentialIssuerProfile_Full
=== PAUSE TestAccCredentialIssuerProfile_Full
=== RUN   TestAccCredentialIssuerProfile_InvalidConfig
=== PAUSE TestAccCredentialIssuerProfile_InvalidConfig
=== RUN   TestAccCredentialType_NewEnv
=== PAUSE TestAccCredentialType_NewEnv
=== RUN   TestAccCredentialType_Full
=== PAUSE TestAccCredentialType_Full
=== RUN   TestAccCredentialType_MetaData
=== PAUSE TestAccCredentialType_MetaData
=== RUN   TestAccCredentialType_CardDesignTemplate
=== PAUSE TestAccCredentialType_CardDesignTemplate
=== RUN   TestAccDigitalWalletApplication_NewEnv
=== PAUSE TestAccDigitalWalletApplication_NewEnv
=== RUN   TestAccDigitalWalletApplication_Full
=== PAUSE TestAccDigitalWalletApplication_Full
=== RUN   TestAccDigitalWalletApplication_InvalidNativeApplication
=== PAUSE TestAccDigitalWalletApplication_InvalidNativeApplication
=== RUN   TestAccDigitalWalletApplication_InvalidAppOpenUrl
=== PAUSE TestAccDigitalWalletApplication_InvalidAppOpenUrl
=== CONT  TestAccCredentialIssuanceRuleDataSource_ByIDFull
=== CONT  TestAccDigitalWalletApplicationsDataSource_NoFilter
=== CONT  TestAccCredentialTypeDataSource_InvalidConfig
=== CONT  TestAccDigitalWalletApplicationDataSource_ByApplicationIDFull
=== CONT  TestAccDigitalWalletApplicationDataSource_NotFound
=== CONT  TestAccCredentialTypesDataSource_NotFound
=== CONT  TestAccCredentialType_Full
=== CONT  TestAccDigitalWalletApplication_InvalidAppOpenUrl
=== CONT  TestAccDigitalWalletApplication_InvalidNativeApplication
=== CONT  TestAccDigitalWalletApplication_Full
=== CONT  TestAccCredentialIssuanceRuleDataSource_ByIDFull
=== CONT  TestAccDigitalWalletApplicationsDataSource_NoFilter
=== CONT  TestAccCredentialTypeDataSource_InvalidConfig
=== CONT  TestAccDigitalWalletApplicationDataSource_ByApplicationIDFull
=== CONT  TestAccDigitalWalletApplicationDataSource_NotFound
=== CONT  TestAccCredentialTypesDataSource_NotFound
=== CONT  TestAccCredentialType_Full
=== CONT  TestAccDigitalWalletApplication_InvalidAppOpenUrl
=== CONT  TestAccDigitalWalletApplication_InvalidNativeApplication
=== CONT  TestAccDigitalWalletApplication_Full
--- PASS: TestAccCredentialTypeDataSource_InvalidConfig (0.34s)
=== CONT  TestAccDigitalWalletApplication_NewEnv
--- PASS: TestAccDigitalWalletApplication_InvalidAppOpenUrl (3.17s)
=== CONT  TestAccCredentialType_CardDesignTemplate
--- PASS: TestAccCredentialType_CardDesignTemplate (0.49s)
=== CONT  TestAccCredentialType_MetaData
--- PASS: TestAccDigitalWalletApplicationDataSource_NotFound (4.07s)
=== CONT  TestAccCredentialIssuerProfileDataSource_NotFound
--- PASS: TestAccCredentialType_MetaData (0.64s)
=== CONT  TestAccCredentialTypeDataSource_NotFound
--- PASS: TestAccCredentialTypeDataSource_NotFound (0.73s)
=== CONT  TestAccCredentialTypeDataSource_ByIDFull
--- PASS: TestAccCredentialTypesDataSource_NotFound (6.80s)
=== CONT  TestAccDigitalWalletApplicationDataSource_ByNameFull
--- PASS: TestAccCredentialIssuanceRuleDataSource_ByIDFull (7.12s)
=== CONT  TestAccDigitalWalletApplicationDataSource_ByIDFull
--- PASS: TestAccDigitalWalletApplicationDataSource_ByApplicationIDFull (7.17s)
=== CONT  TestAccCredentialIssuanceRuleDataSource_InvalidConfig
--- PASS: TestAccCredentialIssuanceRuleDataSource_InvalidConfig (0.28s)
=== CONT  TestAccCredentialIssuerProfileDataSource_ByEnvironmentIDFull
--- PASS: TestAccCredentialIssuerProfileDataSource_NotFound (4.47s)
=== CONT  TestAccCredentialIssuanceRuleDataSource_NotFound
--- PASS: TestAccDigitalWalletApplication_NewEnv (8.23s)
=== CONT  TestAccCredentialIssuerProfile_Full
--- PASS: TestAccDigitalWalletApplication_InvalidNativeApplication (8.85s)
=== CONT  TestAccCredentialType_NewEnv
--- PASS: TestAccCredentialIssuanceRuleDataSource_NotFound (2.50s)
=== CONT  TestAccCredentialIssuerProfile_InvalidConfig
--- PASS: TestAccCredentialIssuerProfile_InvalidConfig (0.25s)
=== CONT  TestAccCredentialIssuanceRule_Full
--- PASS: TestAccDigitalWalletApplicationsDataSource_NoFilter (11.78s)
=== CONT  TestAccCredentialIssuanceRule_InvalidConfigs
--- PASS: TestAccDigitalWalletApplicationDataSource_ByNameFull (5.84s)
=== CONT  TestAccDigitalWalletApplicationsDataSource_NotFound
--- PASS: TestAccCredentialIssuanceRule_InvalidConfigs (0.95s)
=== CONT  TestAccCredentialTypesDataSource_NoFilter
--- PASS: TestAccCredentialTypeDataSource_ByIDFull (7.74s)
--- PASS: TestAccDigitalWalletApplicationDataSource_ByIDFull (6.28s)
--- PASS: TestAccCredentialType_NewEnv (7.39s)
--- PASS: TestAccCredentialIssuerProfileDataSource_ByEnvironmentIDFull (9.41s)
--- PASS: TestAccDigitalWalletApplicationsDataSource_NotFound (7.28s)
--- PASS: TestAccCredentialTypesDataSource_NoFilter (9.73s)
--- PASS: TestAccDigitalWalletApplication_Full (29.92s)
--- PASS: TestAccCredentialType_Full (32.57s)
--- PASS: TestAccCredentialIssuerProfile_Full (35.28s)
--- PASS: TestAccCredentialIssuanceRule_Full (55.28s)
PASS
ok      github.com/pingidentity/terraform-provider-pingone/internal/service/credentials 66.921s
```

### CODE REVIEW

File: docs/resources/credential_type.md [hunk 1]
```diff
@@ -2,15 +2,15 @@
...
+  ~> You must ensure that any fields used in the carddesigntemplate are defined appropriately in metadata.fields or errors occur when you attempt to create a credential of that type.
 ---
 
 # pingone_credential_type (Resource)
 
-Resource to create, read, and update the credential types used by compatible wallet applications.
+Resource to create and manage the credential types used by compatible wallet applications.
 
-~> You must ensure that any fields used in the cardDesignTemplate are defined appropriately in metadata.fields or errors occur when you attempt to create a credential of that type.
+~> You must ensure that any fields used in the card_design_template are defined appropriately in metadata.fields or errors occur when you attempt to create a credential of that type.
```
Comment (patrickcping) [inline docs/resources/credential_type.md:13]: Might be worth putting card_design_template in markdown quotes: `card_design_template`

File: docs/resources/credential_type.md [hunk 2]
```diff
@@ -93,12 +93,12 @@ resource "pingone_credential_type" "verifiedemployee" {
 - `card_design_template` (String) An SVG formatted image containing placeholders for the credentials fields that need to be displayed in the image.
 - `environment_id` (String) PingOne environment identifier (UUID) in which the credential type exists.
 - `metadata` (Attributes) Contains the names, data types, and other metadata related to the credential. (see [below for nested schema](#nestedatt--metadata))
-- `title` (String) Title of the credential. Verification sites are expected to be able to request the issued credential from the compatible wallet app using the title.
+- `title` (String) Title of the credential. Verification sites are expected to be able to request the issued credential from the compatible wallet app using the title.  This value aligns to `${cardTItle}` in the `card_design_template`.
```
Comment (patrickcping) [inline docs/resources/credential_type.md:96]: `${cardTItle}` probably should be `${cardTitle}`

File: internal/service/credentials/resource_credential_type.go [hunk 3]
```diff
@@ -161,18 +173,40 @@ func (r *CredentialTypeResource) Schema(ctx context.Context, req resource.Schema
...
-				Description: "Title of the credential. Verification sites are expected to be able to request the issued credential from the compatible wallet app using the title.",
-				Required:    true,
+				Description:         titleDescription.Description,
+				MarkdownDescription: titleDescription.MarkdownDescription,
+				Required:            true,
 				Validators: []validator.String{
 					stringvalidator.LengthAtLeast(attrMinLength),
+					customstringvalidator.IsRequiredIfRegexMatchesPathValue(
+						regexp.MustCompile(`\${cardTitle}`),
+						"The title argument is required because the ${$cardTitle} element is defined in the card_design_template.",
```
Comment (patrickcping) [inline internal/service/credentials/resource_credential_type.go:183]: `${$cardTitle}` should probably be `${cardTitle}`

Comment (patrickcping) [reply internal/service/credentials/resource_credential_type.go:183]: `card_design_template` should probably be in MD quotes

File: internal/service/credentials/resource_credential_type.go [hunk 4]
```diff
@@ -161,18 +173,40 @@ func (r *CredentialTypeResource) Schema(ctx context.Context, req resource.Schema
...
-				Description: "A description of the credential type.",
-				Optional:    true,
+				Description:         credentialDescription.Description,
+				MarkdownDescription: credentialDescription.MarkdownDescription,
+				Optional:            true,
 				Validators: []validator.String{
 					stringvalidator.LengthAtLeast(attrMinLength),
+					customstringvalidator.IsRequiredIfRegexMatchesPathValue(
+						regexp.MustCompile(`\${cardSubtitle}`),
+						"The description argument is required because the ${$cardSubtitle} element is defined in the card_design_template.",
```
Comment (patrickcping) [inline internal/service/credentials/resource_credential_type.go:202]: `${$cardSubtitle}` => `${cardSubtitle}`

Comment (patrickcping) [reply internal/service/credentials/resource_credential_type.go:202]: card_design_template => `card_design_template`

**Review by patrickcping — CHANGES_REQUESTED** [review]
Small formatting changes

File: docs/resources/credential_type.md
Comment (patrickcping) [inline docs/resources/credential_type.md:1]: These doc entries should probably get a "note" changelog entry, e.g.
https://github.com/pingidentity/terraform-provider-pingone/blob/main/.changelog/378.txt

File: internal/service/credentials/resource_credential_type.go [hunk 5]
```diff
@@ -161,18 +173,40 @@ func (r *CredentialTypeResource) Schema(ctx context.Context, req resource.Schema
...
 
 			"title": schema.StringAttribute{
-				Description: "Title of the credential. Verification sites are expected to be able to request the issued credential from the compatible wallet app using the title.",
-				Required:    true,
+				Description:         titleDescription.Description,
+				MarkdownDescription: titleDescription.MarkdownDescription,
+				Required:            true,
 				Validators: []validator.String{
 					stringvalidator.LengthAtLeast(attrMinLength),
+					customstringvalidator.IsRequiredIfRegexMatchesPathValue(
```
Comment (patrickcping) [inline internal/service/credentials/resource_credential_type.go:181]: This should also probably get a "bug" changelog entry, e.g.
https://github.com/pingidentity/terraform-provider-pingone/blob/35369cf5c82fe9601814c99994d1585e4aedddf7/.changelog/365.txt#L5-L7

**Review by patrickcping — CHANGES_REQUESTED** [review]

File: docs/resources/credential_type.md (hunk 2 above)
Comment (mjspi) [reply docs/resources/credential_type.md:96]: corrected

File: internal/service/credentials/resource_credential_type.go (hunk 3 above)
Comment (mjspi) [reply internal/service/credentials/resource_credential_type.go:183]: fixed

File: internal/service/credentials/resource_credential_type.go (hunk 4 above)
Comment (mjspi) [reply internal/service/credentials/resource_credential_type.go:202]: fixed

File: internal/service/credentials/resource_credential_type.go [hunk 6]
```diff
@@ -118,6 +118,14 @@ func (r *CredentialTypeResource) Schema(ctx context.Context, req resource.Schema
 	const attrMaxPercent = 100
 	const imageMaxSize = 50000
 
+	titleDescription := framework.SchemaAttributeDescriptionFromMarkdown(
+		"Title of the credential. Verification sites are expected to be able to request the issued credential from the compatible wallet app using the title.  This value aligns to `${cardTItle}` in the `card_design_template`.",
```
Comment (patrickcping) [inline internal/service/credentials/resource_credential_type.go:122]: `${cardTItle}` regression

**Review by patrickcping — CHANGES_REQUESTED** [review]
One minor correction

File: internal/service/credentials/resource_credential_type.go (hunk 6 above)
Comment (mjspi) [reply internal/service/credentials/resource_credential_type.go]: I am very annoyed by this.  It's resolved.

**Review by patrickcping — APPROVED** [review]
LGTM 🚀 🚀

**Pull request merged**